atlasdb-cassandra/src/test/java/com/palantir/atlasdb/cassandra/backup/TransactionAborterTest.java

```diff
@@ -0,0 +1,307 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.cassandra.backup;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.KeyspaceMetadata;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Statement;
+import com.datastax.driver.core.TableMetadata;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Range;
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.cassandra.CassandraKeyValueServiceConfig;
+import com.palantir.atlasdb.cassandra.backup.transaction.TransactionTableEntries;
+import com.palantir.atlasdb.cassandra.backup.transaction.TransactionTableEntry;
+import com.palantir.atlasdb.cassandra.backup.transaction.TransactionsTableInteraction;
+import com.palantir.atlasdb.pue.PutUnlessExistsValue;
+import com.palantir.timestamp.FullyBoundedTimestampRange;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TransactionAborterTest {
+    private static final long BACKUP_TIMESTAMP = 123;
+
+    private static final FullyBoundedTimestampRange TIMESTAMP_RANGE =
+            FullyBoundedTimestampRange.of(Range.closed(AtlasDbConstants.STARTING_TS, 200L));
+
+    private static final String TXN_TABLE_NAME = "txn_table";
+    private static final String KEYSPACE = "keyspace";
+
+    @Mock
```

Comment:
I'm twitching from the number of mocks here, but the tests read well!

Reply:
ugh yeah, it's a lot. I can probably try using a real `TransactionsTableInteraction`, which will cut out 5 mocks (we wouldn't need the statements).

The TableMetadata part is dealt with in a nicer way in #5846 , so I could possibly clean that up as part of that PR, which is based off this one, or a follow-up.

Reply:
Hmm, this seems rather involved. I think it would be easier if `TransactionsTableInteraction` was refactored to not use the `CqlSession`, and the `CqlSession` might benefit from less generic methods, so we could call something like `CqlSession.prepareAbortStatement(TransactionsTableInteraction, TableMetadata)`. We could then mock or verify those calls as necessary.

Since it  wouldn't necessarily improve how the tests look (just the number of mocks), I'm less inclined to try this out currently.

Reply:
Yeah, let's keep as is for now